File: LeetCode Solutions/June 2021 LeetCoding Challenge/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.cpp
class Solution {
public:
    int maxArea(int h, int w, vector<int>& horizontalCuts, vector<int>& verticalCuts) {
        sort(horizontalCuts.begin(), horizontalCuts.end());
        sort(verticalCuts.begin(), verticalCuts.end());
        int maxHorizontalCuts = max(horizontalCuts[0], h - horizontalCuts.back()),
            maxVerticalCuts = max(verticalCuts[0], w - verticalCuts.back());
        for (int i=1; i<horizontalCuts.size(); i++){
            maxHorizontalCuts = max(maxHorizontalCuts, horizontalCuts[i] - horizontalCuts[i-1]);
        }
        for (int i=1; i<verticalCuts.size(); i++){
            maxVerticalCuts = max(maxVerticalCuts, verticalCuts[i] - verticalCuts[i-1]);
        }
        return (int)((long)maxHorizontalCuts * maxVerticalCuts % 1000000007);
    }
};

File: LeetCode Solutions/LeetCoding Challenge/June 2021 LeetCoding Challenge/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts/Maximum Area of a Piece of Cake After Horizontal and Vertical Cuts.java
class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        Arrays.sort(horizontalCuts);
        Arrays.sort(verticalCuts);
        int maxHorizontalCuts = Math.max(horizontalCuts[0], h - horizontalCuts[horizontalCuts.length-1]),
            maxVerticalCuts = Math.max(verticalCuts[0], w - verticalCuts[verticalCuts.length-1]);
        for (int i=1; i<horizontalCuts.length; i++){
            maxHorizontalCuts = Math.max(maxHorizontalCuts, horizontalCuts[i] - horizontalCuts[i-1]);
        }
        for (int i=1; i<verticalCuts.length; i++){
            maxVerticalCuts = Math.max(maxVerticalCuts, verticalCuts[i] - verticalCuts[i-1]);
        }
        return (int)((long)maxHorizontalCuts * maxVerticalCuts % 1000000007);
    }
}